examples/jsm/renderers/webgpu/WebGPURenderPipelines.js

```diff
@@ -99,8 +99,9 @@ class WebGPURenderPipelines {
 
 			const disposeCallback = onMaterialDispose.bind( this );
 			materialProperties.disposeCallback = disposeCallback;
+			materialProperties.object = object;
```

Comment:
You assume here that a material is only assigned to a single object. I suggest to not introducing this assumption in the renderer.

Reply:
Yeah. I could add an array but I thought about not complicating it at the moment... just to make the example work that is the focus at the moment.
Work with `object` on top creation of `pipeline`, `materials`, `geometries` can brigh us others problems. 
For this reason I would like change others classes to support even `Multi-Material ` before making a decent implementation here.

Reply:
I'm not sure we want to continue supporting Multi-Materials 🤔

Reply:
Yes, we should consider that. The handling with `groups` was problematic in the past.

Reply:
> Work with object on top creation of pipeline, materials, geometries can brigh us others problems.

I suggest you implement the example without a GUI for now and display the various material all at once like in:

https://threejs.org/examples/webgl_materials